### ISSUE TITLE
Variable names beginning with underscores stopped working

### ISSUE THREAD

**Issue by mrobbert** [TACC/Lmod#587]
**Describe the bug**
We have Lua modules that include variables that begin with and underscore that stopped working after a recent upgrade of Lmod from 8.4 to 8.7.5. As far as I can tell leading underscores are legal in all shells and have been allowed in Lmod previously.
 
**To Reproduce**
Create a modulefile call test.lua with the following single line:

`setenv("_DSM_BARRIER","SHM")
`
Try to load that module and it reports this error:

```
Lmod has detected the following error:  setenv names must be start with a letter followed
letters, numbers and underscores (Not: "_DSM_BARRIER")

While processing the following module(s):
    Module fullname  Module Filename
    ---------------  ---------------
    test             /wendianHome/u/mx/co/mrobbert/mfiles/test.lua
```

**Expected behavior**
The module should load without error and set that variable.

**Desktop (please complete the following information):**
 - OS: Linux
 - Linux distribution: CentOS 7.9
 - Lmod Version: 8.7.5

**Comment by rtmclay** [comment]
Thanks for the clear bug report!  This change was implemented to fix issue #580 but I missed that env. vars. can start with an underscore.  This now works for me.  Please test Lmod 8.7.6 to see if works for you.  Thanks!

**Comment by mrobbert** [comment]
I just updated and it appears to be working now. Thanks!
